### DISCUSSION TITLE
catch_all suppress first item from stdin

### DISCUSSION THREAD

**Discussion by akrigator** [bashly-framework/bashly#509] [Q&A]
Not sure that understand catch_all completely, but seems that something goes wrong with it. I've add a demo command to my_app which (as i guess) should collect all args and std inputs to other_args array
```
commands:
  - name: demo
    catch_all: true
```
but when i call `my_app demo 1 2 3 <<< 4 5 6` it misses 4 in inspect_args output:
```
other_args:
- ${other_args[*]} = 1 2 3 5 6
- ${other_args[0]} = 1
- ${other_args[1]} = 2
- ${other_args[2]} = 3
- ${other_args[3]} = 5
- ${other_args[4]} = 6

```

**Comment by DannyBen** [comment]
> my_app demo 1 2 3 <<< 4 5 6

The use of `<<<` was never advertised as something that should work.
Can you give me a more realistic example of what you are trying to achieve?

Note that you get the exact same behavior if you use this simple bash script:

```bash
#!/usr/bin/env bash
echo "$*"
```

```
$ ./test.sh 1 2 3 <<< 4 5 6
1 2 3 5 6
```

Also note that the first argument to follow  `<<<` can be read with `read`:

```bash
#!/usr/bin/env bash
echo "$*"
read input
echo "$input"
```

```
$ ./test.sh 1 2 3 <<< whatever 4 5 6
1 2 3 4 5 6
whatever

$ ./test.sh 1 2 3 <<< "whatever 4 5 6"
1 2 3
whatever 4 5 6
```

So in regards to bashly - if for some reason you must use `<<<`, you should use `read` in your command code:

```bash
# src/demo_command.sh
inspect_args
read input
echo "Input: $input"
```

**Comment by akrigator** [comment]
@DannyBen thanks for point another direction, yeah problem was in another place
```
➜  ~ arr=(1 2 3)
➜  ~ parallel -n 1 -I% echo % <<< ${arr[*]}
1 2 3
➜  ~ parallel -n 1 -I% echo % ::: ${arr[*]}
1
2
3
```
